$OpenBSD: patch-unico_gtkroundedbox_c,v 1.2 2012/10/04 13:48:28 ajacoutot Exp $

Disable rounded corner code that crashes on newer gtk+3.

Fix segfault in draw_background(GtkThemingEngine *engine, ...)

--- unico/gtkroundedbox.c.orig	Thu Oct  4 15:27:41 2012
+++ unico/gtkroundedbox.c	Thu Oct  4 15:27:44 2012
@@ -85,21 +85,22 @@ gtk_rounded_box_clamp_border_radius (GtkRoundedBox *bo
 }
 
 void
-_gtk_rounded_box_apply_border_radius (GtkRoundedBox    *box,
-                                      GtkThemingEngine *engine,
-                                      GtkStateFlags     state,
-                                      GtkJunctionSides  junction)
+_gtk_rounded_box_apply_border_radius_for_engine (GtkRoundedBox    *box,
+                                                 GtkThemingEngine *engine,
+                                                 GtkStateFlags     state,
+                                                 GtkJunctionSides  junction)
 {
-  GtkCssBorderCornerRadius *corner[4];
+  GtkCssBorderCornerRadius *corner[4] = { NULL, NULL, NULL, NULL };
   guint i;
 
+  /* NOTE: Border radius is disabled due to it not working in GTK+ 3.5 */
   gtk_theming_engine_get (engine, state,
                           /* Can't use border-radius as it's an int for
                            * backwards compat */
-                          "border-top-left-radius", &corner[GTK_CSS_TOP_LEFT],
+                          /*"border-top-left-radius", &corner[GTK_CSS_TOP_LEFT],
                           "border-top-right-radius", &corner[GTK_CSS_TOP_RIGHT],
                           "border-bottom-right-radius", &corner[GTK_CSS_BOTTOM_RIGHT],
-                          "border-bottom-left-radius", &corner[GTK_CSS_BOTTOM_LEFT],
+                          "border-bottom-left-radius", &corner[GTK_CSS_BOTTOM_LEFT],*/
                           NULL);
 
   if (corner[GTK_CSS_TOP_LEFT] && (junction & GTK_JUNCTION_CORNER_TOPLEFT) == 0)
